contracts/control/unstructured/CrossChainOwnable/UCrossChainOwnable.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts/crosschain/CrossChainEnabled.sol";
+import "../UInitializable.sol";
+import "../../../storage/UStorage.sol";
+
+/**
+ * @title UCrossChainOwnable
+ * @notice Library to manage the cross-chain ownership lifecycle of upgradeable contracts.
+ * @dev This contract has been extended from the Open Zeppelin library to include an
+ *      unstructured storage pattern so that it can be safely mixed in with upgradeable
+ *      contracts without affecting their storage patterns through inheritance.
+ */
+abstract contract UCrossChainOwnable is UInitializable, CrossChainEnabled {
+    event OwnerUpdated(address indexed newOwner);
+    event PendingOwnerUpdated(address indexed newPendingOwner);
+
+    error UCrossChainOwnableNotCrossChain();
+    error UCrossChainOwnableNotOwnerError(address sender);
+    error UCrossChainOwnableNotPendingOwnerError(address sender);
+
+    /// @dev The owner address
+    AddressStorage private constant _owner = AddressStorage.wrap(keccak256("equilibria.root.UCrossChainOwnable.owner"));
+    function owner() public view returns (address) { return _owner.read(); }
+
+    /// @dev The pending owner address
+    AddressStorage private constant _pendingOwner = AddressStorage.wrap(keccak256("equilibria.root.UCrossChainOwnable.pendingOwner"));
+    function pendingOwner() public view returns (address) { return _pendingOwner.read(); }
+
+    /**
+     * @notice Initializes the contract setting `owner` as the initial owner
+     * @param owner_ Crosschain owner address
+     */
+    function __UCrossChainOwnable__initialize(address owner_) internal onlyInitializer {
+        if (!_isCrossChain()) revert UCrossChainOwnableNotCrossChain();
+        _updateOwner(owner_);
```

Comment:
I think we should think of something a little smarter here if it's possible - in the original is sets it to the `msg.sender` to guarantee that there's no way you can mess it up. Not exactly sure what that looks like though, we should figure out what the deploy / configure process will generally look like for these.